data/huion-h950p.tablet

```diff
@@ -0,0 +1,24 @@
+# HUION
+# H950P
+# place in /usr/share/libwacom/
+
+[Device]
+Name=Huion H950P
+ModelName=
+DeviceMatch=usb:256c:006d
+Class=Bamboo
+#Width=10
+#Height=6
+IntegratedIn=
+Layout=huion-h610-pro.svg
```

Comment:
A quick google for the 610 and the 950 show that there's a bit of difference in the button layouts. I'd prefer it if we had a custom svg file here instead of re-using the 610. should only be a matter of minutes to create.

Reply:
How would I make an svg file? Is there some tool for it, or something else?

Reply:
inkscape is the reference tool here, there is no custom tool. But from a look at the 610 file I think you can just open that file, move/reshape the center buttons and done.

Reply:
Ok, so I spent a couple hours in Inkscape and haven't managed to make a fully functional svg file, [here's a gif showing what the problem is](https://imgur.com/a/bozreyw). This is my first time using Inkscape btw. 

I duplicated a circle, ungrouped it, swapped out the two center shapes with full circles and moved them, then made sure everything was grouped only to the shapes they needed to be. I don't know what I'm doing wrong. I even started again with a fresh 610 svg file and tried to do the same thing, doing multiple iterations before giving up. 

I do have a question though, how does libwacom know which shapes are linked to which actions?

Reply:
> I do have a question though, how does libwacom know which shapes are linked to which actions?

the buttons have specific IDs, e.g. `ButtonA` and the lines leading to them have `LeaderA` etc. Inkscape shows this on ... object properties? I'd guess you have multiple objects with the same ID?

Reply:
You'd be right about multiple objects having the same ID. I replaced all the circles with manually made rectangles to better match the tablet's buttons, none of them are being highlighted now when the tablet buttons are pressed. I've played around with object IDs, group object IDs, and fill and stroke settings, and I just can't get it to work, trying to match all the settings between the functional 610 file and the non-functional 950 file. Is there any other way that libwacom identifies the shapes?

Reply:
no, it's a bit hard to guess remotely anyway. The [README](https://github.com/linuxwacom/libwacom/blob/master/data/layouts/README) has a full description of what the various names need to be, hopefully this helps you figure out what's wrong. Also, in case this wasn't already obvious: svg files are just xml files so you can grep through them to search for errors.

Reply:
Here it is [https://pastebin.com/nm3JSsGv](https://pastebin.com/nm3JSsGv)

Reply:
It works, all 8 buttons seem to work. I can't thank you enough for figuring out the svg/xml stuff drrwyy. I think it's ready now.